data_steward/cdr_cleaner/cleaning_rules/remove_ehr_data_past_deactivation_date.py

```diff
@@ -0,0 +1,68 @@
+"""
+Ensures there is no EHR data past the deactivation date for deactivated participants.
+
+Original Issue: DC-686
+
+The intent is to sandbox and drop records dated after the date of deactivation for participants
+who have deactivated from the Program.
+"""
+
+# Python imports
+import logging
+
+# Third party imports
+import bq_utils
+
+# Project imports
+import retraction.retract_deactivated_pids as rdp
+import utils.participant_summary_requests as psr
+
+LOGGER = logging.getLogger(__name__)
+
+TICKET_NUMBER = ['dc686']
+
+DEACTIVATED_PARTICIPANTS_COLUMNS = [
+    'participantId', 'suspensionStatus', 'suspensionTime'
+]
+
+
+def remove_ehr_data_queries(project_id, ticket_number, pids_project_id,
+                            pids_dataset_id, tablename):
+    """
+    Creates sandboxes and drops all EHR data found for deactivated participants after
+    their deactivation date
+
+    :param project_id: BQ name of the project
+    :param ticket_number: Jira ticket number to identify and title sandbox tables
+    :param pids_project_id: deactivated participants PIDs table in BQ's project_id
+    :param pids_dataset_id: deactivated participants PIDs table in BQ's dataset_id
+    :param tablename: The name of the table to house the deactivated participant data
+    """
+
+    ehr_union_dataset = bq_utils.get_unioned_dataset_id()
```

Comment:
Is there a reason you are only removing from the unioned dataset?

Reply:
Yes, the ticket says:

Include a check for the `ehr_union` dataset that drops all rows of EHR data dated after the date of deactivation for participants who have deactivated from the Program.

I took that as this rule is only looking at the `ehr_union` dataset which is why I passed it as the dataset to check. Did I interpret that right?

Reply:
Caroline confirmed this (I just wanted to double check!)